### ISSUE TITLE
integrating websockets

### ISSUE THREAD

**Issue by cansado2930** [alexa-js/alexa-app#217]
Hello,
I want to develop a skill in order to connect with a webservice. this webservice works with websockets (wss://) but I have tried to integrate it and Alexa server launchs me an error when I try to start ws. Could you help me about what websockets should I use?
Thanks

**Comment by dblock** [comment]
Your issue has likely nothing to do with Alexa, so I'll close this.

First I would make the WSS code work without the Alexa parts. Once you have you can post your next issue in integrating with alexa-app here. Also, websockets are typically a connection that is maintained for the duration of the process, so I am not sure how that fits with the whole alexa request/response flow.

**Comment by ericblade** [comment]
Also, "launchs me an error" isn't particularly helpful, if you're asking for help, please post details of what modules you're using, and what the actual errors are.

**Comment by cansado2930** [comment]
Thanks for your answers and Sorry, I should have explained more details. I had the code working in a basic nodejs file of wss. And then I added it to Alexa-app and then Alexa-server said me that there are an error and the app couldn't be loaded. But it doesn't said me what error. So I tried other wss modules and all have the same problem. And in addition, how @dblock have said, I know that wss breaks the flow of Alexa with request/response so I have my question of it would work or not. But there is an example (Alexa-sdk) with kodi that uses websockets, so I deduced that it is more or less possible. And How I couldn't get more information about error, I asked you.
After a lot try and error, I discovered that I have to wait a small time in order to use the object that create "var ws= new WebSocket("URI");" because it is undefined yet when nodejs executes the next line (ws.on("conected", function(){....});) and curiously the same code on basic nodejs file works correctly. The module that I'm using is "ws". So I'm going to integrate messages from websocket with promises in order to try to fit Alexa request/responde flow.
Thanks and sorry again.
